src/main/java/org/springframework/data/hazelcast/repository/query/GeoPredicate.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.hazelcast.repository.query;
+
+import static com.hazelcast.query.impl.IndexUtils.canonicalizeAttribute;
+
+import java.util.Map;
+
+import org.springframework.data.geo.Distance;
+import org.springframework.data.geo.Metric;
+import org.springframework.data.geo.Metrics;
+import org.springframework.data.geo.Point;
+
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.Extractable;
+/**
+ * Geo Predicate - Used to calculate near and within queries
+ * <li>Finds all the Points within the given distance range from source Point.
+ * <li>Finds all the Points within given Circle.
+ *
+ * @param <K> key of map entry
+ * @param <V> value of map entry
+ * @author Ulhas R Manekar
+ */
+public class GeoPredicate<K, V>
+        implements Predicate<K, V> {
+
+    private static final long serialVersionUID = 1L;
+    private static final double KM_TO_MILES = 0.621371;
+    private static final double KM_TO_NEUTRAL = 0.539957;
+    private static final double R = 6372.8;
+
+    final String attributeName;
+    final Point queryPoint;
+    final Distance distance;
+
+    /**
+     * Constructor accepts the name of the attribute which is of type Point.
+     * Constructs a new geo predicate on the given point
+     *
+     * @param attribute    the name of the attribute in a object within Map which is of type Point.
+     *
```

Comment:
Nit: formatting, remove empty lines here.

Reply:
Done, checked in.